cyclonedx/model/definition.py

```diff
@@ -15,21 +15,391 @@
+        officially supported in the standard without having to use additional namespaces or create extensions.
+
+        Return:
+            Set of `Property`
+        """
+        return self._properties
+
+    @properties.setter
+    def properties(self, properties: Iterable[Property]) -> None:
+        self._properties = SortedSet(properties)
+
+    @property
+    @serializable.xml_array(serializable.XmlArraySerializationType.NESTED, 'reference')
+    @serializable.xml_sequence(8)
+    def external_references(self) -> 'SortedSet[ExternalReference]':
+        """
+        Provides the ability to document external references related to the component or to the project the component
+        describes.
+
+        Returns:
+            Set of `ExternalReference`
+        """
+        return self._external_references
+
+    @external_references.setter
+    def external_references(self, external_references: Iterable[ExternalReference]) -> None:
+        self._external_references = SortedSet(external_references)
+
+
+@serializable.serializable_class
+class Level:
+    """
+    Level of compliance for a standard.
+    """
+
+    def __init__(
+        self, *,
+        bom_ref: Optional[Union[str, BomRef]] = None,
+        identifier: Optional[str] = None,
+        title: Optional[str] = None,
+        description: Optional[str] = None,
+        requirements: Optional[Iterable[Union[str, BomRef]]] = None,
+    ) -> None:
+        self._bom_ref = bom_ref_from_str(bom_ref)
+        self.identifier = identifier
+        self.title = title
+        self.description = description
+        self.requirements = requirements or []  # type:ignore[assignment]
+
+    def __lt__(self, other: Any) -> bool:
+        if isinstance(other, Level):
+            return (_ComparableTuple((self.bom_ref, self.identifier))
+                    < _ComparableTuple((other.bom_ref, other.identifier)))
+        return NotImplemented
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, Level):
+            return hash(other) == hash(self)
+        return False
+
+    def __hash__(self) -> int:
+        return hash((
+            self.bom_ref, self.identifier, self.title, self.description, tuple(self.requirements)
+        ))
+
+    def __repr__(self) -> str:
+        return f'<Level bom-ref={self.bom_ref}, identifier={self.identifier}, title={self.title}, ' \
+               f'description={self.description}>'
+
+    @property
+    @serializable.json_name('bom-ref')
+    @serializable.type_mapping(BomRefHelper)
+    @serializable.xml_attribute()
+    @serializable.xml_name('bom-ref')
+    def bom_ref(self) -> BomRef:
+        """
+        An optional identifier which can be used to reference the level elsewhere in the BOM.
+        Every bom-ref MUST be unique within the BOM.
+
+        If a value was not provided in the constructor, a UUIDv4 will have been assigned.
```

Comment:
>         If a value was not provided in the constructor, a UUIDv4 will have been assigned.

has this been tested? i dont see any code giving those unset `bom-ref`s a (temp) value ...

Reply:
Introduced some additional tests for this in `cyclonedx/model/definition.py`